### PR TITLE
add observatorium 2.3 and main branch to prow

### PR DESCRIPTION
**Pull request by haoqing0110**
add observatorium 2.3 and main branch to prow
https://github.com/open-cluster-management/backlog/issues/7260

### CODE REVIEW

File: ci-operator/jobs/open-cluster-management/observatorium/open-cluster-management-observatorium-release-2.3-postsubmits.yaml [hunk 1]
```diff
@@ -53,3 +53,41 @@ postsubmits:
...
+    spec:
+      containers:
+      - args:
+        - -f
+        - /opt/build-harness/Makefile.prow
+        - osci/publish
+        command:
+        - make
+        env:
+        - name: COMPONENT_NAME
```
Comment (clyang82) [inline ci-operator/jobs/open-cluster-management/observatorium/open-cluster-management-observatorium-release-2.3-postsubmits.yaml:71]: s/COMPONENT_NAME/OSCI_COMPONENT_NAME/

**Review by clyang82 — APPROVED** [review]

**Review by clyang82 — APPROVED** [review]

**Review by petr-muller — APPROVED** [review]

**Pull request merged**